### PR TITLE
Validation and model spec

### PR DESCRIPTION
**Pull request by Hassaanjbaig-code**
### In this pull request 

- Add the following validations:
    -  For the [User model](https://github.com/microverseinc/curriculum-rails/blob/fd2a839fca24d77bd85d73df0d7a76668188530f/blog-app/blog_app_validations_model_spec.md?plain=1#L27-L28)
    - For the [Post model](https://github.com/microverseinc/curriculum-rails/blob/fd2a839fca24d77bd85d73df0d7a76668188530f/blog-app/blog_app_validations_model_spec.md?plain=1#L30-L33)
 - Add a unit test for all models

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    expect(subject.bio).to eq('Teacher from Bangla.')
+  end
+  it 'subject name should not be empty' do
+    subject.name = ''
+    expect(subject).to_not be_valid
+  end
+  it 'subject postscounter should be grater than 0' do
+    subject.postscounter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (tresorsawasawa) [inline spec/models/user_spec.rb]: - [x] As required for the project to have tests to any custom methods in the models, kindly make sure you also add tests to the `#recent_posts` method in the `User` model.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    post.commentscounter = 1
+
+    expect(post.commentscounter).to be >= 0
+  end
+
+  it 'likes_counter should be greater than or equal to zero' do
+    post.likescounter = 1
+
+    expect(post.likescounter).to be >= 0
+  end
```
Comment (tresorsawasawa) [inline spec/models/post_spec.rb]: - [x] As required for the project to have tests to any custom methods in the models, kindly make sure also add tests to the `#recent_comments` and `update_author_posts_counter` methods in the `Post` model.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Hassaanjbaig-code 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  describe 'update_post_comments_counter' do
+    it 'should update post commentscounter after creating a new Comment' do
+      Comment.create(author: @user, post: @post, text: 'This is my first comment')
+      expect(@post.commentscounter).to eq(1)
+    end
+    it 'should return the text' do
+      comment = Comment.create(author: @user, post: @post, text: 'This is my first comment')
+      expect(comment.text).to eq('This is my first comment')
+    end
+  end
```
[the remark this answers is not in the document]
Comment (Hassaanjbaig-code) [reply spec/models/comment_spec.rb:20]: @tresorsawasawa Hello, how are you? I just wanted to say #update_the_comment_counter is a private method but you can see In a `comment_spec,rb` in lines # 11-14 I write the test 
When the user adds a comment it updates the commentscounter as you can see in the code
This is also for the like_spec.rb 

Kindle again check it

Thank you

**Review by Deepakdanger — APPROVED** [review]
Hi @Hassaanjbaig-code, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**